test/unit/helpers/navigation_items_helper_test.rb

```diff
@@ -0,0 +1,60 @@
+require "test_helper"
+
+class NavigationItemsHelperTest < ActionView::TestCase
+  context "NavigationItemsHelper" do
+    should "include editor-only links when user is an editor" do
+      assert_equal [
+        { text: "Publications", href: root_path, active: false },
+        { text: "Add artefact", href: new_artefact_path, active: false },
+        { text: "Downtime", href: downtimes_path, active: false },
+        { text: "Reports", href: reports_path, active: false },
+        { text: "Search by user", href: user_search_path, active: false },
+        { text: "Sign out", href: "/auth/gds/sign_out" },
+      ], navigation_items(is_editor: true)
+    end
+
+    should "not include editor-only links when user is not an editor" do
+      assert_equal [
+        { text: "Publications", href: root_path, active: false },
+        { text: "Reports", href: reports_path, active: false },
+        { text: "Search by user", href: user_search_path, active: false },
+        { text: "Sign out", href: "/auth/gds/sign_out" },
+      ], navigation_items(is_editor: false)
+    end
+
+    should "not include user name link if user has name" do
+      assert_equal [
+        { text: "Publications", href: root_path, active: false },
+        { text: "Reports", href: reports_path, active: false },
+        { text: "Search by user", href: user_search_path, active: false },
+        { text: "Name", href: Plek.new.external_url_for("signon") },
+        { text: "Sign out", href: "/auth/gds/sign_out" },
+      ], navigation_items(is_editor: false, user_name: "Name")
+    end
+
+    should "apply set link to active if page is current link" do
```

Comment:
Should we be checking the user name link in this test too?

Reply:
the name isn't added to the list of links if it isn't set and I didn't provide it in this test